File: src/main/java/com/openpms/common/R.java
package com.openpms.common;

import lombok.Data;
import org.springframework.http.ResponseEntity;

@Data
public class R {
    private int code;
    private String msg;

    public R(int code, String msg) {
        this.code = code;
        this.msg = msg;
    }

    public static ResponseEntity<?> Ok() {
        return ResponseEntity.ok(new R(0, "ok"));
    }

    public static ResponseEntity<?> Ok(String msg) {
        return ResponseEntity.ok(new R(0, msg));
    }

    public static ResponseEntity<?> Fail(int code, String msg) {
        return ResponseEntity.status(400).body(new R(code, msg));
    }

    public static ResponseEntity<?> WithStatus(int code, String msg, int httpCode) {
        return ResponseEntity.status(httpCode).body(new R(code, msg));
    }
}

File: src/main/kotlin/com/creator/common/R.kt
package com.creator.common

import org.springframework.http.ResponseEntity

class R(private val code: Int, private val msg: String) {
    companion object {
        fun Ok(): ResponseEntity<*> {
            return ResponseEntity.ok<R>(R(0, "ok"))
        }

        fun Ok(msg: String): ResponseEntity<*> {
            return ResponseEntity.ok<R>(R(0, msg))
        }

        fun Fail(code: Int, msg: String): ResponseEntity<*> {
            return ResponseEntity.status(400).body<R>(R(code, msg))
        }

        fun WithStatus(code: Int, msg: String, httpCode: Int): ResponseEntity<*> {
            return ResponseEntity.status(httpCode).body<R>(R(code, msg))
        }
    }
}
